hydra-main/src/main/java/com/addthis/hydra/job/web/resources/JobsResource.java

```diff
@@ -862,6 +868,61 @@ public Response getRevisions(@QueryParam("id") String id,
         }
     }
 
+    // http://localhost:5052/job/f641cc5a-e3bf-48c4-a533-c6f35045b45c/log?out=1&lines=10&node=0&runsAgo=0
+
+    @GET
+    @Path("/{jobID}/log")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getJobTaskLog(@PathParam("jobID") String jobID,
+                                  @QueryParam("lines") @DefaultValue("50") int lines,
+                                  @QueryParam("runsAgo") @DefaultValue("0") int runsAgo,
+                                  @QueryParam("out") @DefaultValue("1") int out,
+                                  @QueryParam("minion") String minion,
+                                  @QueryParam("port") String port,
+                                  @QueryParam("node") String node) {
+        JSONObject body = new JSONObject();
+        boolean error = false;
+        try {
+            if (minion == null) {
+                body.put("error", "Missing required query parameter 'minion'");
+                return Response.status(Response.Status.BAD_REQUEST).entity(body.toString()).build();
+            } else if (node == null) {
+                body.put("error", "Missing required query parameter 'node'");
+                return Response.status(Response.Status.BAD_REQUEST).entity(body.toString()).build();
+            } else if (port == null) {
+                body.put("error", "Missing required query parameter 'node'");
+                return Response.status(Response.Status.BAD_REQUEST).entity(body.toString()).build();
+            } else {
+                String url = String.format("http://%s:%s/job.log?id=%s&node=%s&runsAgo=%d&lines=%d&out=%d",
+                        minion, port, jobID, node, runsAgo, lines, out);
```

Comment:
Personally I'd use a more expressive API, such as the `URIBuilder` and its `setParameter` method, rather than building the url manually.